map_creator/src/load_capability_map.cpp

```diff
@@ -0,0 +1,142 @@
+#include <ros/ros.h>
+
+#include "map_creator/capability.h"
+
+
+#include "H5Cpp.h"
+#include <hdf5.h>
+
+using namespace H5;
+using namespace std;
+
+#define CAP_DATASETNAME "capability_dataset"
+#define CAP_GROUPNAME "/Capability"
+#define RANK_OUT 2
+
+class loadHD5{
+public:
+
+void h5ToMultiMapCap(const hid_t dataset, multimap<vector<double>, vector<double> >& sphereColor){
+
+    hsize_t dims_out[2], count[2], offset[2], dimsm[2];
+    hid_t dataspace = H5Dget_space (dataset);    /* dataspace handle */
+    int rank      = H5Sget_simple_extent_ndims (dataspace);
+    herr_t status_n  = H5Sget_simple_extent_dims (dataspace, dims_out, NULL);
+    herr_t status;
+    offset[0] = 0;
+    offset[1] = 0;
+    count[0]  = dims_out[0];
+    count[1]  = 10;
+    double data_out[count[0]][count[1]];
+    status = H5Sselect_hyperslab (dataspace, H5S_SELECT_SET, offset, NULL, 
+                                  count, NULL);
+    dimsm[0] = count[0];
+    dimsm[1] =  count[1];
+    hid_t memspace;
+    memspace = H5Screate_simple (RANK_OUT, dimsm, NULL);  
+    status = H5Dread (dataset, H5T_NATIVE_DOUBLE, memspace, dataspace,
+                      H5P_DEFAULT, data_out);
+    for(int i=0;i<count[0];i++){
+	vector<double> posAndQuat;
+	vector<double> enumRIAngle;
+        for(int j=0;j<2;j++)
+	    {
+		 enumRIAngle.push_back(data_out[i][j]);
+	    }
+	for(int j=2;j<9;j++)
+	    {
+		 posAndQuat.push_back(data_out[i][j]);
+	    }
+        for(int j=9;j<10;j++)
+	    {
+		 enumRIAngle.push_back(data_out[i][j]);
+	    }
+        
+        sphereColor.insert(pair<vector<double>, vector<double> >(posAndQuat, enumRIAngle));
+    }
+  }
+};
+
+
+int main(int argc, char **argv)
+{
+    if (argc<2){
+	ROS_ERROR_STREAM("Please provide the name of the Capability map. If you have not created it yet, Please create the map by running the create cpability map launch file in map_creator");
+	return 1;
+    } 
+    else{
+	
+    ros::init(argc, argv, "workspace");
+    ros::NodeHandle n;
+    ros::Publisher workspace_pub = n.advertise<map_creator::capability>("capability_map", 1);
+    ros::Rate loop_rate(10);
+  
+    int count = 0;
+    while (ros::ok())
```

Comment:
@akgoins Moved the while loop down. Don't prefer latched publisher for now, though it is a better solution. The visualization has some issues that it does not update the changes instantly. We have to reset the rviz window for the changes to take effect. Because some of the functions are not Q_SLOTS. We have to create them as Q_SLOTS. This can be considered as an issue, which I surely solve later.